vpr/src/server/telegrambuffer.cpp

```diff
@@ -0,0 +1,79 @@
+#ifndef NO_SERVER
+
+#include "telegrambuffer.h"
+
+namespace comm {
+
+void TelegramBuffer::append(const ByteArray& bytes) {
+    m_raw_buffer.append(bytes);
+}
+
+bool TelegramBuffer::check_telegram_header_presence() {
+    std::size_t signature_start_index = m_raw_buffer.find_sequence(TelegramHeader::SIGNATURE, TelegramHeader::SIGNATURE_SIZE);
+    if (signature_start_index != std::size_t(-1)) {
```

Comment:
I think this was mentioned by Soheil before, but casting -1 to size_t is a bit strange. A standard optional would be more appropriate.

Reply:
this was fixed 
https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/2526/commits/5886d611e60a7760c288e72b463103fde00fc256
Note: i avoid using std::optional for ByteArray.h, because std""optional is c++17 and i had issue with building openfpga with it, where the C++ standard is lower. If i am not mistaken. Sorry i don't remember details it was quite a long time ago.